Newline at the beginning of the file throws an exception

**Describe the bug**
```
SEVERE: Internal error: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
java.util.concurrent.CompletionException: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1(CancelTokens.scala:38)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1$adapted(CancelTokens.scala:36)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.charAt(String.java:658)
	at scala.collection.immutable.StringOps$.apply$extension(StringOps.scala:41)
	at scala.meta.internal.metals.MetalsEnrichments$XtensionString.lastIndexBetween(MetalsEnrichments.scala:411)
	at scala.meta.internal.metals.MultilineStringFormattingProvider.pipeInScope(MultilineStringFormattingProvider.scala:86)
	at scala.meta.internal.metals.MultilineStringFormattingProvider.$anonfun$withToken$1(MultilineStringFormattingProvider.scala:131)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	... 4 more
```
**To Reproduce**
Steps to reproduce the behavior:
1. Go to any file
2. Move cursor to the start of the file
3. Press enter
4. See error

**Expected behavior**
Nothing happens.

**Installation:**
- Operating system: macOS/Windows/Linux
- Editor: Visual Studio Code/Atom/Vim/Sublime/Emacs
- Metals version: v0.8.1

**Search terms**
on type formatting exception

--- a/metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala
+++ b/metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala
@@ -78,10 +78,13 @@ final class MultilineStringFormattingProvider(
       newlineAdded: Boolean
   ): Boolean = {
     val newLineBeforePos =
-      text.lastIndexBetween('\n', upperBound = pos.start - 1)
+      text.lastIndexBetween('\n', upperBound = Math.max(pos.start - 1, 0))
     val pipeSearchStop =
       if (newlineAdded)
-        text.lastIndexBetween('\n', upperBound = newLineBeforePos - 1)
+        text.lastIndexBetween(
+          '\n',
+          upperBound = Math.max(newLineBeforePos - 1, 0)
+        )
       else newLineBeforePos
     val lastPipe = text.lastIndexBetween('|', pipeSearchStop, pos.start - 1)
     lastPipe > pipeSearchStop

--- a/tests/unit/src/test/scala/tests/OnTypeFormattingSuite.scala
+++ b/tests/unit/src/test/scala/tests/OnTypeFormattingSuite.scala
@@ -4,6 +4,22 @@ import munit.Location
 
 class OnTypeFormattingSuite extends BaseLspSuite("onTypeFormatting") {
 
+  // Ensures that entering a newline at the beginning of a file doesn't
+  // throw an exception
+  // https://github.com/scalameta/metals/issues/1469
+  check(
+    "top-of-file",
+    s"""
+       |@@
+       |object Main {}
+       |""".stripMargin,
+    s"""
+       |
+       |
+       |object Main {}
+       |""".stripMargin
+  )
+
   check(
     "correct-string",
     s"""

